.github/workflows/CI.yml

```diff
@@ -25,14 +25,70 @@ jobs:
         components: rustfmt
     - run: cargo fmt --all --check
 
+  build-protoc-plugin:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macOS-latest, windows-latest]
+    outputs:
+      cache-hit: ${{ steps.cache-plugin.outputs.cache-hit }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Cache protoc plugin
+        id: cache-plugin
+        uses: actions/cache@v4
+        with:
+          path: ${{ runner.temp }}/protoc-plugin
+          # The key changes only when plugin source files change
+          key: ${{ runner.os }}-protoc-plugin-${{ hashFiles('protoc-gen-rust-grpc/src/**', 'protoc-gen-rust-grpc/.bazelrc', 'protoc-gen-rust-grpc/MODULE.bazel') }}
+      - name: Install Bazel
+        if: steps.cache-plugin.outputs.cache-hit != 'true'
+        uses: bazel-contrib/setup-bazel@0.15.0
+        with:
+          # Avoid downloading Bazel every time.
+          bazelisk-cache: true
+          # Store build cache per workflow.
+          disk-cache: ${{ github.workflow }}
+          # Share repository cache between workflows.
+          repository-cache: true
+          module-root: ./protoc-gen-rust-grpc
+      - name: Build protoc plugin
```

Comment:
Does this step take a long time? Generally, speaking I would rather that we build the plugin in each test job to ensure we are building the latest code and there is no caching issues. Could you explain the reasoning for breaking it up like this?